launcher/ui/dialogs/ExportMrPackDialog.cpp

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (C) 2023 TheKodeToad <TheKodeToad@proton.me>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "ExportMrPackDialog.h"
+#include "ui/dialogs/CustomMessageBox.h"
+#include "ui/dialogs/ProgressDialog.h"
+#include "ui_ExportMrPackDialog.h"
+
+#include <QFileDialog>
+#include <QFileSystemModel>
+#include <QJsonDocument>
+#include <QMessageBox>
+#include "FileSystem.h"
+#include "MMCZip.h"
+#include "modplatform/modrinth/ModrinthPackExportTask.h"
+
+ExportMrPackDialog::ExportMrPackDialog(InstancePtr instance, QWidget* parent)
+    : QDialog(parent), instance(instance), ui(new Ui::ExportMrPackDialog)
+{
+    ui->setupUi(this);
+    ui->name->setText(instance->name());
+    ui->summary->setText(instance->notes().split(QRegularExpression("\\r?\\n"))[0]);
+
+    auto model = new QFileSystemModel(this);
+    // use the game root - everything outside cannot be exported
+    QString root = instance->gameRoot();
+    proxy = new FileIgnoreProxy(root, this);
+    proxy->setSourceModel(model);
+
+    QDir::Filters filter(QDir::AllEntries | QDir::NoDotAndDotDot | QDir::AllDirs | QDir::Hidden);
+
+    for (QString file : QDir(root).entryList(filter)) {
+        if (!(file == "mods" || file == "coremods" || file == "datapacks" || file == "config" || file == "options.txt" ||
+              file == "servers.dat"))
+            proxy->blockedPaths().insert(file);
+    }
+
+    QDir modsIndex(instance->gameRoot() + "/mods/.index");
+    if (modsIndex.exists())
+        proxy->blockedPaths().insert("mods/.index");
```

Comment:
```suggestion
    auto mc_instance = dynamic_cast<MinecraftInstance*>(instance.get());
    if (mc_instance && mc_instance->loaderModList()) {
        auto modsIndex = mc_instance->loaderModList()->indexDir();
        if (modsIndex.exists()) {
            QDir root_dir{ root };
            proxy->blockedPaths().insert(root_dir.relativeFilePath(index_dir.absoluteFIlePath());
        }
    }
```

ideally we'd not duplicate hardcoded folder names across the codebase, like in this case.

Reply:
Isn't `&& mc_instance->loaderModList()` redundant since the pointer is always reset if it's null?